crates/revm/src/db/utils.rs

```diff
@@ -0,0 +1,22 @@
+use tokio::runtime::{Handle, Runtime};
+
+// Hold a tokio runtime handle or full runtime
+#[derive(Debug)]
+pub(crate) enum HandleOrRuntime {
+    Handle(Handle),
+    Runtime(Runtime),
+}
+
+impl HandleOrRuntime {
+    #[inline]
+    pub(crate) fn block_on<F>(&self, f: F) -> F::Output
+    where
+        F: std::future::Future + Send,
+        F::Output: Send,
+    {
+        match self {
+            Self::Handle(handle) => tokio::task::block_in_place(move || handle.block_on(f)),
+            Self::Runtime(rt) => rt.block_on(f),
```

Comment:
this will panic if called within async execution context

Reply:
Yes, that’s intended. Passing a runtime mostly would be the sync code. If user is within async execution context, they should use new directly instead.

Reply:
As stated above, if user is in async context, they should call new as #1557 suggests. If user is in sync context, they should provide a runtime, mostly current_thread runtime.

If user intends to mix different contexts, they should be careful by themselves as it is known to be bad practice and cause problems hard to tackle.